record/record_test.go

```diff
@@ -326,30 +341,69 @@ func writeRecords(records [][]byte) ([]byte, error) {
 			return nil, err
 		}
 	}
+
 	if err := w.Close(); err != nil {
 		return nil, err
 	}
-	return buf.Bytes(), nil
+
+	results.buf = buf.Bytes()
+	return results, nil
 }
 
-func TestBasicRecover(t *testing.T) {
-	records := [][]byte{
-		[]byte(strings.Repeat("a", blockSize-headerSize)),
-		[]byte(strings.Repeat("b", blockSize-headerSize)),
-		[]byte(strings.Repeat("c", blockSize-headerSize)),
+// Corrupt the checksum of the record that starts the the specified block
+// offset. The number of the block offset is 0 based.
+func corruptBlock(buf []byte, blockNum int) {
+	// Ensure we always permute at least 1 byte of the checksum.
+	if buf[blockSize*blockNum+0] == 0x00 {
+		buf[blockSize*blockNum+0] = 0xff
+	} else {
+		buf[blockSize*blockNum+0] = 0x00
 	}
-	buf, err := writeRecords(records)
+
+	buf[blockSize*blockNum+1] = 0x00
+	buf[blockSize*blockNum+2] = 0x00
+	buf[blockSize*blockNum+3] = 0x00
+}
+
+func TestRecoverNoOp(t *testing.T) {
+	testData, err := makeTestRecords(
```

Comment:
I had it like (records [][]byte, buf []byte, retErr error) but I changed it because it felt clumsy to pass around the [][]byte and []byte as I refactored. I can put it back but this felt like it was cleaner to me.